src/app/pages/cinemahall/cinemahall.component.ts

```diff
@@ -0,0 +1,64 @@
+import { Component, OnInit } from '@angular/core';
+import { Observable } from 'rxjs';
+import { mergeRequest } from 'src/app/servises/hint/request';
+import { concatResponses } from 'src/app/servises/hint/response';
+import { MediaResult } from 'src/app/servises/model/response-data';
+import { PopularMediaService } from 'src/app/servises/popular-media.service';
+import { TrendingMediaService } from 'src/app/servises/trending-media.service';
+import { UpcomingMediaService } from 'src/app/servises/upcoming-media.service';
+
+interface Sections {
+  title: string,
+  content: MediaResult[]
+}
+
+type sectionType = 'popular' | 'trending' | 'upcoming'
+
+@Component({
+  selector: 'mov-cinemahall',
+  templateUrl: './cinemahall.component.html',
+  styleUrls: ['./cinemahall.component.scss']
+})
+export class CinemahallComponent implements OnInit {
+  constructor(
+    private readonly apiPopularMedia: PopularMediaService,
+    private readonly apiTrendingMedia: TrendingMediaService,
+    private readonly apiUpcomingMedia: UpcomingMediaService
+  ){
+
+    mergeRequest<MediaResult[]>([{type: 'movie'}, {type: 'tv'}], this.apiPopularMedia.requestMediaByType.bind(this.apiPopularMedia))
```

Comment:
Looks too complicated, discuss direct forkJoin using.

Reply:
simplified a litle

Reply:
Still I think that `mergeRequest` function is not needed, but let's leave it for now.